mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,660 @@
+import copy
+import math
+import os.path as osp
+
+import mmcv
+import numpy as np
+from torch.nn.modules.utils import _pair
+
+from mmaction.localization.proposal_utils import temporal_iou
+from .base import BaseDataset
+from .registry import DATASETS
+from .utils import (load_localize_proposal_file, parse_directory,
+                    process_localize_proposal_list)
+
+
+class SSNInstance(object):
+    """Proposal instance of SSN.
+
+    Args:
+        start_frame (int): Index of the proposal's start frame.
+        end_frame (int): Index of the proposal's end frame.
+        num_frames (int): Total frames of the video.
+        label (int): Classification of the proposal.
+        best_iou (int): The highest IOU with the groundtruth instance.
+        overlap_self (int): Percent of the proposal's own span contained
+            in a groundtruth instance.
+    """
+
+    def __init__(self,
+                 start_frame,
+                 end_frame,
+                 num_frames,
+                 label=None,
+                 best_iou=None,
+                 overlap_self=None):
+        self.start_frame = start_frame
+        self.end_frame = min(end_frame, num_frames)
+        self.num_frames = num_frames
+        self.label = label if label is not None else -1
+        self.coverage = (end_frame - start_frame) / num_frames
+        self.best_iou = best_iou
+        self.overlap_self = overlap_self
+        self.loc_reg = None
+        self.size_reg = None
+        self.regression_targets = [0., 0.]
+
+    def compute_regression_targets(self, gt_list, fg_thresh):
+        """Compute regression targets of positive proposals.
+
+        Args:
+            gt_list (list): The list of groundtruth instances.
+            fg_thresh (float): Minimum threshold of overlap of
```

Comment:
@dreamerlin what's our standard abbreviation for threshold

Reply:
threshold. btw, I think fg is a little bit confusing, should we use the full name `foreground` or add a comment to explain it ?

Reply:
I vote for using the full name `foreground_threshold`. Same to `foreground_pool` etc.